internal/services/inference/deployment.go

```diff
@@ -177,14 +186,19 @@ func ResourceDeploymentCreate(ctx context.Context, d *schema.ResourceData, m int
 		return diag.FromErr(err)
 	}
 
+	modelID := d.Get("model_id")
+	if !scwvalidation.IsUUID(modelID.(string)) {
+		modelID = regional.ExpandID(modelID).ID
+	}
+
 	req := &inference.CreateDeploymentRequest{
-		Region:    region,
-		ProjectID: d.Get("project_id").(string),
-		Name:      d.Get("name").(string),
-		NodeType:  d.Get("node_type").(string),
-		ModelName: d.Get("model_name").(string),
-		Tags:      types.ExpandStrings(d.Get("tags")),
-		Endpoints: buildEndpoints(d),
+		Region:       region,
+		ProjectID:    d.Get("project_id").(string),
+		Name:         d.Get("name").(string),
+		NodeTypeName: d.Get("node_type").(string),
+		ModelID:      modelID.(string),
```

Comment:
You can directly use `ModelID: locality.ExpandID(d.Get("model_id").(string))` which handles both raw and regional IDs